lib/token/asymmetric.go

```diff
@@ -0,0 +1,238 @@
+package token
```

Comment:
I really would like for this to be labelled asymmetric_nacl or something, because this essentially is the multiencoder's interface over a asymmetric nacl